web3/contract.py

```diff
@@ -722,6 +776,161 @@ def __prepared_function(self, **kwargs):
         return getattr(contract_modifier_func(modifier_dict), self.__function)
 
 
+class ContractMethod(object):
+    """
+    """
+    contract = None
+    method_name = None
+    web3 = None
+    transaction = None
```

Comment:
This doesn't appear to be used anywhere.

Reply:
Its used in the factory method on [line 924](https://github.com/ethereum/web3.py/pull/481/files#diff-2171ba22a13b6793b04c4a1287049a67R924) where the kwargs are checked against the ContractMethod attributes before they are passed to type().